ProjectShop/UserApp/views.py

```diff
@@ -1,3 +1,22 @@
 from django.shortcuts import render
+from rest_framework import generics
+from rest_framework.permissions import IsAuthenticated, IsAdminUser
+
+from UserApp.serializers import UserSerializer
+from UserApp.models import User
+
+
+class UserRestView(generics.RetrieveAPIView, generics.ListAPIView):
+    """View of one user model object"""
+    queryset = User.objects.all()
+    serializer_class = UserSerializer
+    lookup_url_kwarg = 'user_id'
+    permission_classes = {IsAuthenticated, IsAdminUser}
```

Comment:
I think you should use a list instead of a set.

Reply:
Agree with you. Changed sets on tuples in last commit.